out/.gitignore

```diff
@@ -0,0 +1,3 @@
+*
+!/.gitignore
```

Comment:
what does this line and the next line mean?

Reply:
! unignores what's been ignored

Reply:
Ok, so the `!/test/` line isn't doing anything then?

Reply:
this is a standard thing, see eg: https://stackoverflow.com/questions/115983/how-can-i-add-an-empty-directory-to-a-git-repository/932982#932982

once something is already in the index, ignoring it in .gitignore (eg via `*` or by commenting out the `!` entry after `*`) should have no effect, maybe that's why you're not noticing the difference in this particular case. Try on an empty repo

Reply:
Based on what I understand and from the stack overflow post you gave, this line and the next line don't do anything once this file is added.  I'd say let's remove these lines, unless I'm missing something.

Reply:
discussed offline https://dlang.slack.com/archives/D8QUL983T/p1521574536000103 and removed the !/test/ but kept !.gitignore